sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp

```diff
@@ -121,6 +197,14 @@ namespace Azure { namespace Core { namespace Test {
       return std::make_unique<T>(url, *credential, options);
     }
 
+    template <class T> T InitClientOptions()
+    {
+      // Run instrumentation before creating the client
+      T options;
+      PrepareOptions(options);
+      return options;
+    }
+
```

Comment:
The idea is to enable InitClientOptions() to be called without creating the options first. Like:

auto client = BlobClient(url, credential, InitClientOptions<BlobClientOption>());

otherwise, the pattern would be
BlobClientOptions options;
InitClientOptions(options);  // The name would change to `PreparaClientOptions`
auto client = BlobClient(url, credential, options);